web/src/app/users/UserContactMethodList.js

```diff
@@ -34,60 +37,133 @@ const testCM = gql`
   }
 `
 
-export default class UserContactMethodList extends React.PureComponent {
-  static propTypes = {
-    userID: p.string.isRequired,
-    readOnly: p.bool,
-  }
+const useStyles = makeStyles({
+  actionGrid: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+})
+
+export default function UserContactMethodList(props) {
+  const classes = useStyles()
+
+  const [showVerifyDialogByID, setShowVerifyDialogByID] = useState(null)
+  const [showEditDialogByID, setShowEditDialogByID] = useState(null)
+  const [showDeleteDialogByID, setShowDeleteDialogByID] = useState(null)
+  const [testID, setTestID] = useState(null)
+
+  const [sendTest] = useMutation(testCM, {
+    variables: {
+      id: testID,
```

Comment:
If sendTest accepts that as a parameter we can, but when defining the variables in the mutation's initialization, a hook can't be nested inside of a second function. I can check the apollo docs.

Reply:
If a second parameter is specified, `useEffect` will only run if the variable given in that parameter has changed.

Reply:
Looks like that is unsupported:

```
UserContactMethodList.js:99 - Network error: HTTP Response 422: Unprocessable Entity
{"errors":[{"message":"cannot be null","path":["variable","id"]}],"data":null}
```

Reply:
I know it technically works, but it's not super clear what is happening.

If we can't pass in args (their documentation looks like it _should_ work though), then maybe we can move the `useMutation` to a separate component that's fed a CM ID directly so we can still fire the mutation on-click instead of post-render via `useEffect`.

![image](https://user-images.githubusercontent.com/595010/61241730-b71a2900-a709-11e9-92a7-ac0dad7722f5.png)

That's an odd type signature... I'll see if I can figure out locally why it wouldn't be using the options param

Reply:
I think it will be more clear as more people start using hooks (since doing something in a hook based off a variable rather than a prop I can see as a common use case).

I can either add a comment to the `useEffect` we define, or as you're saying create a sort of higher order component that wraps it, if preferred.

Reply:
This seems to work fine:
![image](https://user-images.githubusercontent.com/595010/61243115-df575700-a70c-11e9-92dc-93af87cddbf3.png)